.github/CONTRIBUTING.md

```diff
@@ -167,3 +167,71 @@ after the starting description.
  * ...
  */
 ```
+
+### Error Messages
+
+User-facing error messages should be clear, concise, and consistent. Error
+messages should be in sentence case but should not end with a period. Error
+messages should be free of grammatical errors and typos and written in American
+English.
+
+> ⚠️ Note that the error message style guide is a work in progress, and not all
+> the error messages have been updated to conform to the current styles. Error
+> message styles that should be followed:
+
+1. Messages should start with an upper case:
+
+```
+Bad: cannot parse input
+Good: Cannot parse input
+```
+
+2. Messages should not end with a period:
+
+```
+Bad: Cannot parse input.
+Good: Cannot parse input
+```
+
+3. Message should use quotes for values:
+
+```
+Bad: Cannot parse input x
+Good: Cannot parse input "x"
+```
+
```

Comment:
Thanks for updating. But I think this rule needs more consideration. For example, using quotes makes sense when the value is a string, but it probably makes more sense to not use quotes when the value is a number. I guess there are edge cases where it's difficult to decide which pattern to use (nested object, etc).

For now, I'd suggest removing this rule (until we come up with a good way to formulate this)

```suggestion
```

Reply:
I think quoting strings is sensible… and I've often used direct JSON serializations (e.g. ``const message = "…found `${JSON.stringify(str)}`…";``) in my own error messages to get quotes for free (and representations that can be copy+pasted back into some JS source). The downside to that is when it's desirable to print literals which are replaced by escaped sequences in the output (e.g. line breaks).

Reply:
This is why I attempted to updated the error messages in a folder, I think this gives us some good concrete examples to talk about and see how they _feel_.

I'm happy to remove this rule, or we could also use this:

3. Message should use quotes for values for strings:

```
Bad: Cannot parse input hello, world
Good: Cannot parse input "hello, world"
Good: Attempting to grow buffer by 10 bytes, which would exceed the maximum size of 100 bytes
```

What do you think, do you want to keep this rule for Strings?

Reply:
> 3. Message should use quotes for values for strings:
> 
> ```
> Bad: Cannot parse input hello, world
> Good: Cannot parse input "hello, world"
> Good: Attempting to grow buffer by 10 bytes, which would exceed the maximum size of 100 bytes
> ```

This looks good to me